app/models/header_lookup.rb

```diff
@@ -10,24 +10,57 @@ class HeaderLookup
   MESH_ID_URI = "https://id.nlm.nih.gov/mesh/"
   MEMOIZED_MESH_FILE = "memoized_mesh.txt"
   MEMOIZED_LCSH_FILE = "memoized_lcsh.txt"
+  SEARCHABLE_MESH_FILE = "subjects_mesh.yml"
+  SEARCHABLE_LCSH_FILE = "subjects_lcsh.yml"
 
-  def initialize(memoized_mesh_file_path=MEMOIZED_MESH_FILE, memoized_lcsh_file_path=MEMOIZED_LCSH_FILE)
-    @@memoized_mesh ||= read_memoized_headers(memoized_mesh_file_path)
-    @@memoized_lcsh ||= read_memoized_headers(memoized_lcsh_file_path)
+  def initialize
+    # these are the terms to search through for header lookups
+    @@searchable_mesh_terms ||= YAML.load_file(SEARCHABLE_MESH_FILE)
+    @@searchable_lcsh_terms ||= YAML.load_file(SEARCHABLE_LCSH_FILE)
+
+    # these are values that have been previously found from the searchable terms
+    @@memoized_mesh ||= read_memoized_headers(MEMOIZED_MESH_FILE)
+    @@memoized_lcsh ||= read_memoized_headers(MEMOIZED_LCSH_FILE)
   end
 
   def pid_lookup_by_scheme(term="", scheme="")
     if term.blank? || scheme.blank?
       nil
     elsif scheme == :mesh
-      @@memoized_mesh[term] || mesh_term_pid_lookup(term)
+      @@memoized_mesh[term] || mesh_term_pid_local_lookup(term) || mesh_term_pid_lookup(term)
     elsif scheme == :lcsh
-      @@memoized_lcsh[term] || lcsh_term_pid_lookup(term)
+      @@memoized_lcsh[term] || lcsh_term_pid_local_lookup(term) || lcsh_term_pid_lookup(term)
     else
       nil
     end
   end
 
+  def mesh_term_pid_local_lookup(mesh_term="")
+    @@searchable_mesh_terms.each do |term_json|
+      if term_json["subject"].downcase == mesh_term.downcase
+        mesh_id = term_json["id"]
+        @@memoized_mesh[mesh_term] = mesh_id
+        File.write(MEMOIZED_MESH_FILE, @@memoized_mesh)
+        return mesh_id
+      end
+    end
+
+    nil
+  end
+
+  def lcsh_term_pid_local_lookup(lcsh_term="")
+    @@searchable_lcsh_terms.each do |term_json|
+      if term_json["subject"].downcase == lcsh_term.downcase
+        lcsh_id = term_json["id"]
+        @@memoized_lcsh[lcsh_term] = lcsh_id
+        File.write(MEMOIZED_LCSH_FILE, @@memoized_lcsh)
+        return lcsh_id
+      end
+    end
+
+    nil
```

Comment:
if nothing is found, return nil so that the `||` statement will continue on to use a web api lookup